### DISCUSSION TITLE
reviewdog による自動レビューについて

### DISCUSSION THREAD

**Discussion by potappo** [mozilla-japan/translation#736] [General]
今日のミートアップで自分担当の Pull Request を見ていて、reviewdog による自動レビューが気になったので、自分の調査、Slack、報告会での参加者とのやり取りでわかったことを共有します。

# 自動レビューの仕組み
MDN のレポジトリーに Pull Request をすると、GitHub Action が動き、色々なチェックが行われます。
チェックでエラーになってマージできないということは、以前からあったと思いますが、エラーになった場合に、そのエラーに対する修正の提案を自動で行うのが reviewdog です。 reviewdog は修正内容の投稿をするだけで、エラーチェック自体は別のツールによって行われます。MDN では、markdownlint と prettier によるエラーチェックがあります（Action の "Lint and review content files”）。

# 気になったエラー修正（余分に改行されてしまう）
展開する操作が面倒と思うので画像で貼り付けます。
![image](https://github.com/mozilla-japan/translation/assets/17092516/3c538264-762a-41a7-8e6d-d4fa8c84eacb)
元 PR は以下です。
https://github.com/mdn/translated-content/pull/17917
実際のエラーの Actionログは以下にあります。
https://github.com/mdn/translated-content/actions/runs/7540000163/job/20523677196
改行されてしまうのは、該当部分が HTML として扱われていることが原因です。以下に解説があります。
- [Prettierの設定オプションについてまとめてみた](https://zenn.dev/rescuenow/articles/c07dd571dfe10f#htmlwhitespacesensitivity-html-%E7%A9%BA%E7%99%BD%E3%81%AE%E6%84%9F%E5%BA%A6)
- [Prettier の HTML Whitespace Sensitivity について](https://sosukesuzuki-blog.netlify.app/2vav3KVwgiq0U7ihWq3wwC)

不自然な自動レビューによる修正提案が頻発するようなら困ると思い、調べ始めたのですが、ひとまずは様子見とします。
自動レビューで困っていることや対処法などあれば、このディスカッションにお寄せください。

**Comment by mfuji09** [comment]
とりあえず、該当部分のコードブロックの頭を、 ` ```html` から ` ```html-nolint` に変更すると、このリンターによるチェックおよび自動修正を回避することができます。

この問題はこれだけではなく、HTMLを日本語訳したときに、日本語中に現れる英数字の前後に空白を入れた場合、その空白が強制的に改行に置き換えられてしまうことがあるなど、特に日本語のような単語間に空白を入れない言語では、この対応によりソースが読みにくくなることかま多いと思います。

そもそも、MDNの日本語訳では英語版のHTMLをコピーして、部分的にローカライズするだけだと思いますので、あまりリンターで検査・修正する必要性を感じません。むしろ、勝手に改行位置が変更されることがあるので、英語版が更新されたときに、それに合わせて翻訳版を更新するのが面倒になると感じています。

Prettierを導入した人に抗議を申し入れたのですが、このツールは私たちが開発しているものではないから、との一点張りで取り合ってもらえませんでした。

そもそも、特定の文字数で改行を入れて行を折り返すという発想自体、様々なスクリーンサイズがある現代としては、時代に逆行するとても古い発想だと思います。
